include/deal.II/base/mpi.h

```diff
@@ -1500,14 +1500,26 @@ namespace Utilities
       AssertIndexRange(root_process, n_procs);
       (void)n_procs;
 
-      std::vector<char> buffer      = Utilities::pack(object_to_send, false);
-      unsigned int      buffer_size = buffer.size();
+      std::vector<char> buffer;
+      unsigned int      buffer_size;
```

Comment:
I think for some MPI implementations, valgrind is not able to detect whether the variable `buffer_size` has been initialized. I suggest to put
```suggestion
      unsigned int      buffer_size = numbers::invalid_unsigned_int;
```